tests/unit/test_credentials.py

```diff
@@ -972,5 +973,137 @@ def test_source_profile_does_not_exist(self):
             provider.load()
 
 
+class TestSamlGenericFormsBasedAuthenticator(unittest.TestCase):
+    authenticator = credentials.SamlGenericFormsBasedAuthenticator()
```

Comment:
OK. By default I'll move this instance to setUp(). For the other 3, they are just constant definition, so I'll leave them here. Let me know if you think otherwise.